snapcraft/tests/test_commands_pull.py

```diff
@@ -0,0 +1,120 @@
+        super().make_snapcraft_yaml(self.yaml_template.format(parts=parts))
+        open('icon.png', 'w').close()
+
+        parts = []
+        for i in range(n):
+            part_dir = os.path.join(common.get_partsdir(), 'pull{}'.format(i))
+            state_file = os.path.join(part_dir, 'state')
+            parts.append({
+                'part_dir': part_dir,
+                'state_file': state_file,
+            })
+
+        return parts
+
+    def test_pull_invalid_part(self):
+        fake_logger = fixtures.FakeLogger(level=logging.ERROR)
+        self.useFixture(fake_logger)
+        self.make_snapcraft_yaml()
+
+        with self.assertRaises(EnvironmentError) as raised:
+            pull.main(['no-pull', ])
+
+        self.assertEqual(
+            raised.exception.__str__(),
+            "The part named 'no-pull' is not defined in 'snapcraft.yaml'")
+
+    def test_pull_defaults(self):
+        fake_logger = fixtures.FakeLogger(level=logging.ERROR)
+        self.useFixture(fake_logger)
+        parts = self.make_snapcraft_yaml()
+
+        pull.main()
+
+        self.assertTrue(os.path.exists(common.get_partsdir()),
+                        'Expected a parts directory')
+        self.assertTrue(os.path.exists(parts[0]['part_dir']),
+                        'Expected a part directory for the pull0 part')
+        self.assertTrue(os.path.exists(parts[0]['state_file']),
+                        'Expected a state file for the pull0 part')
+
+        with open(parts[0]['state_file']) as sf:
+            state = sf.readlines()
+        self.assertEqual(len(state), 1, 'Expected only one line in the state '
+                         'file for the pull0 part')
+        self.assertEqual(state[0], 'pull', 'Expected the state file for pull0 '
+                         'to be \'pull\'')
+
+    def test_pull_one_part_only_from_3(self):
+        fake_logger = fixtures.FakeLogger(level=logging.ERROR)
+        self.useFixture(fake_logger)
+        parts = self.make_snapcraft_yaml(n=3)
+
+        pull.main(['pull1', ])
+
+        self.assertTrue(os.path.exists(common.get_partsdir()),
+                        'Expected a parts directory')
+        self.assertTrue(os.path.exists(parts[1]['part_dir']),
+                        'Expected a part directory for the pull1 part')
+        self.assertTrue(os.path.exists(parts[1]['state_file']),
+                        'Expected a state file for the pull1 part')
+
+        with open(parts[1]['state_file']) as sf:
+            state = sf.readlines()
+        self.assertEqual(len(state), 1, 'Expected only one line in the state '
+                         'file for the pull1 part')
+        self.assertEqual(state[0], 'pull', 'Expected the state file for pull1 '
+                         'to be \'pull\'')
+
+        for i in [0, 2]:
+            self.assertFalse(os.path.exists(parts[i]['part_dir']),
+                             'Expected for only to be a part dir for pull1')
```

Comment:
This is hard to read. I'm not sure what are you trying to say.
Maybe something like: "Pulled wrong part."